package/MDAnalysis/core/AtomGroup.py

```diff
@@ -2230,23 +2232,94 @@ def packIntoBox(self, box=None, inplace=True):
         .. versionadded:: 0.8
 
         """
+        from ..core import distances
+
         if box is None:  #Try and auto detect box dimensions
             box = self.dimensions  # Can accept any box
 
         if box.shape == (3, 3):
-            if (box.diagonal() == 0.0).any():  # for a vector representation, diagonal cannot be zero
```

Comment:
I think this just catches if the Timestep allocated the unitcell without filling with values, so it should be exactly zero.